fapolicy_analyzer/ui/faprofiler.py

```diff
@@ -0,0 +1,273 @@
+# Copyright Concurrent Technologies Corporation 2021
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+import os
+import pwd
+import re
+import subprocess
+import time
+from .fapd_manager import FapdMode
+from datetime import datetime as DT
+from enum import Enum
+
+
+class ProfSessionStatus(Enum):
+    QUEUED = 0
+    INPROGRESS = 1
+    COMPLETED = 2
+
+
+def demote(enable, user_uid, user_gid):
+    def result():
+        os.setgid(user_gid)
+        os.setuid(user_uid)
+    if enable:
+        return result
+    else:
+        return None
+
+
+def comma_delimited_kv_string_to_dict(string_in):
+    """Generates dictionary from comma separated string of k=v pairs"""
+    d = None
+    if string_in:
+        d = {p[0]: re.sub(r'^"|"$', "", p[1])
+             for p in [s.split("=")
+                       for s in [c.strip()
+                                 for c in string_in.split(",")]]}
+    return d
```

Comment:
I think this will give you want you want without all the nesting.

```suggestion
    if not string_in:
        return None
    return {
        k: v.strip('"')
        for k, v in dict(x.strip().split("=") for x in string_in.split(",")).items()
    }
```

Also I'd suggest for both this and the `demote` methods to either make them  private members of the `FaProfSession` class or at least name them with the leading single underscore to indicate they are not methods intended for use outside this module.

Reply:
Thank you. I was hunting for a better dict comprehension, but always only came up with something that was only 5-10 chars shorter, so I went with the easiest to read. Your's is much, much slicker.

Reply:
Done. Push shortly after testing.